solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -451,22 +451,24 @@ public NamedList<Object> request(SolrRequest<?> solrRequest, String collection)
     final ResponseParser parser =
         solrRequest.getResponseParser() == null ? this.parser : solrRequest.getResponseParser();
 
+    Throwable abortCause = null;
     try {
       InputStreamResponseListener listener = new InputStreamResponseListener();
       req.send(listener);
       Response response = listener.get(idleTimeout, TimeUnit.MILLISECONDS);
       InputStream is = listener.getInputStream();
       assert ObjectReleaseTracker.track(is);
-
       return processErrorsAndResponse(solrRequest, parser, response, is);
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
+      abortCause = e;
       throw new RuntimeException(e);
     } catch (TimeoutException e) {
```

Comment:
it's not clear to me when to capture aboreCause and when not.  Here, you chose not to on purpose?  Why?

Reply:
I missed it here. I think it may not be needed, because TimeoutExceptions, from what I can see, already abort the request, but it's no harm to have the abort in this case too, worst case it'll be a noop.